gen/ir/template_helpers.go

```diff
@@ -79,7 +79,32 @@ func (t Type) uriFormat() string {
 	return t.EncodeFn()
 }
 
+// externalType defines what available interface to use for encoding/decoding.
+func (t Type) externalType(e ExternalEncoding) ExternalEncoding {
+	switch {
+	case e.Has(ExternalNative):
+		return ExternalNative
+	case e.Has(ExternalJSON):
+		return ExternalJSON
+	case e.Has(ExternalText):
+		return ExternalText
+	case e.Has(ExternalBinary) && t.Schema.Type == jsonschema.String &&
+		(t.Schema.Format == "byte" || t.Schema.Format == "base64"):
+		return ExternalBinary
+	default:
+		return 0
+	}
+}
```

Comment:
This function decides which of the available interfaces to use. Not sure if we should amend the order based on the caller though i.e. should we prefer `(Un)MarshalText` for URL params?

Reply:
Done. See following comment for reasoning behind new order: https://github.com/ogen-go/ogen/pull/1460#pullrequestreview-2993546473